### DISCUSSION TITLE
New Dev Working With Cypress. Code Review appreciated!

### DISCUSSION THREAD

**Discussion by Sedafox** [cypress-io/cypress#22667] [Questions and Help]
Good afternoon! My name is Seda, and I am a manual Quality Assurance tester. My dream is to be an Automation Engineer, and that's how I found Cypress! This is my first project, designed to give me experience on how to automate different scenarios I might find on the web. Any feedback is appreciated, and I hope I am on the right track! 

https://github.com/Sedafox/ProjectAgaru

**Comment by portyu9** [comment]
Hi @Sedafox 

I took a look through Project Agaru, and yes — you are definitely on the right track.

For a first Cypress project, you’ve covered a strong variety of real browser scenarios, and I like that you’re already using Page Objects, beforeEach(), and Cypress assertions instead of relying on hard-coded waits.

Three simple improvements I’d recommend next:

Centralize the base URL. You currently repeat http://localhost:7080 throughout the project. Add it as Cypress baseUrl, then use relative paths such as cy.visit('/login'). This makes the framework cleaner and easier to maintain.
Strengthen your assertions. Make sure every test proves the expected result, not just performs the actions. For example, after logging out, verify that the login page or expected logout message appears.
Separate test data from page behavior. Move reusable values such as usernames, passwords, and other test data out of Page Object methods and into fixtures or Cypress configuration. This keeps the Page Objects focused on application behavior and makes tests easier to maintain.

Those three changes are relatively small, but they will make the project noticeably more maintainable and closer to how professional automation frameworks are structured.

You’re absolutely moving in the right direction — keep building on it rather than starting over. 👍

**Reply by portyu9** [reply]
@Sedafox Please mark this answer as accepted if it helped, thank you.
